core/server/TaskWorker.cpp

```diff
@@ -133,12 +135,22 @@ void TaskWorker::workerThread(LogContext log_context) {
             break;
           }
         }
+
+        db_connection_backoff = 1;
       } catch (TraceException &e) {
         err_msg = e.toString();
+
         DL_ERROR(log_context, "Task worker "
                                   << id() << " exception: " << err_msg
                                   << " task_id is " << m_task->task_id
                                   << " cmd is " << cmd);
+
+        if (err_msg.find("SDMS DB interface failed") != std::string::npos) {
+            int sleep_time = db_connection_backoff;
+            db_connection_backoff = min(db_connection_backoff * 2, 60);
```

Comment:
Add info or debug log including the backoff time, you need to include which task.